### PR TITLE
[fix] Invalid `CarrierFields` structure

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

This library was failing to deserialize a `CarrierAccount` from the API due to a failed deserialization of `CarrierFields`.

The previous assumption was that the API is returning "fields" in the JSON data for a `CarrierAccount` like so:

```json
{
  "id": "ca_123",
  "object": "CarrierAccount",
  ...
  "fields": {
    "credentials": [
      {
        "key": "account_number",
        "visibility": "visible",
        "label": "DHL Account Number",
        "value": "123456"
      },
      {
        "key": "country",
        "visibility": "visible",
        "label": "Account Country Code (2 Letter)",
        "value": "US"
      }
    ]
  }
}
```

When in fact, the API is returning "fields" like so:

```json
{
  "id": "ca_123",
  "object": "CarrierAccount",
  ...
  "fields": {
    "credentials": {
      "account_number": {
        "visibility": "visible",
        "label": "DHL Account Number",
        "value": "123456"
      },
      "country": {
        "visibility": "visible",
        "label": "Account Country Code (2 Letter)",
        "value": "US"
      }
    }
  }
}
```

Specifically, the "credentials" key under "fields" is not a list of dictionaries, but a dictionary of dictionaries, with the type of each credential field being the key in the key-value pair.

This PR updates the `CarrierFields` object structure to match the JSON schema.

Because the key for each type of credential field can by dynamic, users will have to access each credential field via a dictionary index:

```csharp
string countryLabel = myCarrierAccount.Fields.Credentials["country"].Label
```

This PR also adds the missing "logo" property for a `CarrierAccount`.

# Testing

- Add unit test to confirm deserialization works as expected
  - The carrier accounts used in our existing unit tests do not have fields returned by the API, so the serialization issue was not previously caught.

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: EasyPost/Models/API/CarrierFields.cs [hunk 1]
```diff
@@ -14,7 +15,7 @@ public class CarrierFields : EasyPostObject
         ///     The credentials used in the production environment.
         /// </summary>
         [JsonProperty("credentials")]
-        public CarrierField? Credentials { get; set; }
+        public Dictionary<string, CredentialsField>? Credentials { get; set; }
```
Comment (Justintime50) [inline EasyPost/Models/API/CarrierFields.cs:18]: issue: I don't like the name change here (even if the new name may be more correct or readable). It moves this from a bug fix to a breaking change which we should avoid. We can fix the underlying deserialization without users needing to alter their code aside from a version bump.

**Review by Justintime50 — CHANGES_REQUESTED** [review]

File: EasyPost/Models/API/CarrierFields.cs (hunk 1 above)
Comment (nwithan8) [reply EasyPost/Models/API/CarrierFields.cs:18]: To be fair, because of this deserialization issue, a user has never been able to interact with a `CarrierField` object before, since every attempt to create one would have failed.

Comment (Justintime50) [reply EasyPost/Models/API/CarrierFields.cs:18]: Correct, though again, this breaking change will require end-users to make changes in their code if they've already deployed this - even if today it's not working. Let's not major version bump over something that should just be a bug fix patch release.

**Review by Justintime50 — APPROVED** [review]

File: EasyPost/Models/API/CarrierFields.cs [hunk 2]
```diff
@@ -47,12 +48,6 @@ public class CarrierField : EasyPostObject
     {
         #region JSON Properties
 
-        /// <summary>
-        ///     The key of the field.
-        /// </summary>
-        [JsonProperty("key")]
-        public string? Key { get; set; }
-
```
Comment (jchen293) [inline EasyPost/Models/API/CarrierFields.cs:55]: Why is this removed?

Comment (nwithan8) [reply EasyPost/Models/API/CarrierFields.cs:55]: Because it does not exist in the response JSON schema. Instead, the "key" is the actual key in the key-value dictionaries.

**Review by jchen293 — APPROVED** [review]
